File: src/main/java/boj/problems/step2/No2480.java
package boj.problems.step2;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.StringTokenizer;

public class No2480 {
    public static void main(String[] args) throws IOException {
        BufferedReader input = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter output = new BufferedWriter(new OutputStreamWriter(System.out));

        output.write(solve(input) + "\n");

        input.close();
        output.flush();
        output.close();
    }

    static int solve(BufferedReader input) throws IOException {
        StringTokenizer st = new StringTokenizer(input.readLine());
        int a = Integer.parseInt(st.nextToken());
        int b = Integer.parseInt(st.nextToken());
        int c = Integer.parseInt(st.nextToken());

        return getPrizeMoney(a, b, c);
    }

    private static int getPrizeMoney(int a, int b, int c) {
       if (a == b && b == c) {
           return 10000 + (a * 1000);
       }

       if (a == b || a == c) {
           return 1000 + (a * 100);
       }

       if (b == c) {
           return 1000 + (b * 100);
       }

       return getMaximum(a, b, c) * 100;
    }

    private static int getMaximum(int a, int b, int c) {
        int max = a;

        if (b > max) {
            max = b;
        }
        if (c > max) {
            max = c;
        }

        return max;
    }
}

File: src/main/kotlin/boj/problems/step2/No2480.kt
package boj.problems.step2

import java.io.*
import java.util.*

object No2480 {
    @Throws(IOException::class)
    @JvmStatic
    fun main(args: Array<String>) {
        val input = BufferedReader(InputStreamReader(System.`in`))
        val output = BufferedWriter(OutputStreamWriter(System.out))
        output.write(
            """
                ${solve(input)}
                
                """.trimIndent()
        )
        input.close()
        output.flush()
        output.close()
    }

    @Throws(IOException::class)
    fun solve(input: BufferedReader): Int {
        val st = StringTokenizer(input.readLine())
        val a = st.nextToken().toInt()
        val b = st.nextToken().toInt()
        val c = st.nextToken().toInt()
        return getPrizeMoney(a, b, c)
    }

    private fun getPrizeMoney(a: Int, b: Int, c: Int): Int {
        if (a == b && b == c) {
            return 10000 + a * 1000
        }
        if (a == b || a == c) {
            return 1000 + a * 100
        }
        return if (b == c) {
            1000 + b * 100
        } else getMaximum(a, b, c) * 100
    }

    private fun getMaximum(a: Int, b: Int, c: Int): Int {
        var max = a
        if (b > max) {
            max = b
        }
        if (c > max) {
            max = c
        }
        return max
    }
}
